runner/src/artifacts.rs

```diff
@@ -124,7 +124,37 @@ pub fn locate_artifact(name: &str) -> Option<Artifact> {
         "Could not find 'firmware' directory"
     );
 
-    // Check if one entry match the name
+    // Look for artifact inside benchmark folder
+    let artifact = find_artifact(&firmware_path, name);
+    if artifact.is_some() {
+        return artifact;
+    }
+
+    // Get the path to the firmware/benchmark directory
+    firmware_path.push("benchmark");
+    assert!(
+        firmware_path.is_dir(),
+        "Could not find 'firmware/benchmark' directory"
+    );
+
+    // Look for artifact inside benchmark folder
+    let artifact = find_artifact(&firmware_path, name);
+    if artifact.is_some() {
+        return artifact;
+    }
+
+    // Else check if the artifact is defined in the manifest
+    let external = get_external_artifacts();
+    if let Some(artifact) = external.get(name) {
+        return Some(artifact.clone());
+    }
+
+    // Could not find artifact
+    None
+}
+
+/// Check if one entry match the name
+fn find_artifact(firmware_path: &PathBuf, name: &str) -> Option<Artifact> {
```

Comment:
Let's call it `find_artifact_source`. There are multiple kinds of artifacts (sources or binaries), here we only look for sources that we can compile.